### ISSUE TITLE
Renovate incorrectly updates release candidate Gradle versions

### ISSUE THREAD

**Issue by charleskorn** [renovatebot/renovate#7890]
**What Renovate type, platform and version are you using?**

GitHub app

**Describe the bug**

I'm currently using Gradle 6.8 release candidate 1. I enabled Renovate on my repository and it raised [this PR](https://github.com/batect/batect/pull/752) to "upgrade" to Gradle 6.7.1 (the latest stable version).

There are two issues with the PR:

1. It tried to downgrade the version of Gradle in use
2. It incorrectly updated `gradle-wrapper.properties`, leaving it in an invalid state:

```diff
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.8-rc-1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.8-rc-6.7.1-all.zip
```

(the correct value for `distributionUrl` for v6.7.1 would be `https\://services.gradle.org/distributions/gradle-6.7.1-all.zip`)

**Relevant debug logs**

The relevant job is https://app.renovatebot.com/dashboard#github/batect/batect/270170764 - I don't know why, but the logs won't load for me.

**To Reproduce**

The repo affected by the PR (https://github.com/batect/batect) reproduces the issue.

**Comment by viceice** [comment]
I think it's because we only extract version `6.8` and that doesn't exist, so renovate tries to downgrade.

And because of wrong version extract, the downgrade fails.

**Comment by Baneeishaque** [comment]
![g1](https://user-images.githubusercontent.com/6133009/106804822-e5fd4480-668b-11eb-88d6-79569e4a2333.PNG)
![g2](https://user-images.githubusercontent.com/6133009/106804829-e8f83500-668b-11eb-92c6-7a3e58c68d08.PNG)

**Comment by rarkins** [comment]
Can someone provide a minimal public reproduction of this?

**Comment by charleskorn** [comment]
https://github.com/batect/batect/pull/752 is the original example of this issue. 

You should be able to create a new Gradle project with `gradle init` and then replace `gradle-wrapper.properties` with the base version in that PR to reproduce the problem.

**Comment by rarkins** [comment]
Please see https://github.com/renovatebot/renovate/blob/master/docs/development/minimal-reproductions.md

**Comment by charleskorn** [comment]
https://github.com/charleskorn/renovate-gradle-prerelease-issue is a minimal reproduction of the issue.

**Comment by MaronHatoum** [comment]
not reproducible, reproduction repo: https://github.com/MaronHatoum/renovate-gradle-prerelease-issue/pulls

updated `gradle-wrapper.properties` file:

![image](https://user-images.githubusercontent.com/98313426/182832995-0770e0b5-1e7d-4e7d-8a54-4df3503c5eaa.png)

![image](https://user-images.githubusercontent.com/98313426/182833027-dcef3dce-9b82-4cd5-ab5c-e68a082cfdfd.png)


- I think we can close this issue

**Comment by viceice** [comment]
- probably fixed by #11454
